Failed tests (11192): TestReplicateAfterTruncation

The following test appears to have failed:

[#11192](https://circleci.com/gh/cockroachdb/cockroach/11192):

```
I0113 08:22:21.001775 923 storage/replica.go:1031  raft: proposing ADD_REPLICA {2 2 2} for range 1
W0113 08:22:21.006074 923 kv/dist_sender.go:635  failed to invoke BeginTransaction [/Local/Range/RangeDescriptor/"",/Min), ConditionalPut [/Local/Range/RangeDescriptor/"",/Min), Put [/Meta2/Max,/Min), Put [/Meta1/Max,/Min), EndTransaction [/Local/Range/RangeDescriptor/"",/Min): unexpected value: raw_bytes:"\340u\323\345\003\010\001\022\000\032\001\223\"\006\010\001\020\001\030\001(\002" timestamp:<wall_time:0 logical:42 > 
I0113 08:22:21.006690 923 client/db.go:485  failed batch: unexpected value: raw_bytes:"\340u\323\345\003\010\001\022\000\032\001\223\"\006\010\001\020\001\030\001(\002" timestamp:<wall_time:0 logical:42 > 
I0113 08:22:21.008837 923 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
I0113 08:22:21.009034 923 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- FAIL: TestReplicateAfterTruncation (0.06s)
	client_raft_test.go:483: storage/replica_command.go:1737: change replicas of 1 failed: unexpected value: raw_bytes:"\340u\323\345\003\010\001\022\000\032\001\223\"\006\010\001\020\001\030\001(\002" timestamp:<wall_time:0 logical:42 > 
=== RUN   TestStoreRangeUpReplicate
E0113 08:22:21.015205 923 gossip/gossip.go:210  different node IDs were set for the same gossip instance (2147483647, 1)
I0113 08:22:21.015977 923 raft/raft.go:446  [group 1] 1 became follower at term 5
I0113 08:22:21.016265 923 raft/raft.go:234  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I0113 08:22:21.016460 923 raft/raft.go:526  [group 1] 1 is starting a new election at term 5
I0113 08:22:21.017143 923 raft/raft.go:459  [group 1] 1 became candidate at term 6
I0113 08:22:21.017254 923 raft/raft.go:508  [group 1] 1 received vote from 1 at term 6
I0113 08:22:21.017376 923 raft/raft.go:487  [group 1] 1 became leader at term 6
I0113 08:22:21.020877 923 storage/replica.go:1611  gossiping cluster id  from store 1, range 1
--
I0113 08:22:44.115645 923 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestLogSplits (1.14s)
	<autogenerated>:28: prior leak detected, leaktest disabled
=== RUN   Example_rebalancing
--- PASS: Example_rebalancing (1.19s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	31.375s
=== RUN   TestBatchBasics
I0113 08:22:52.206013 998 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.01s)
=== RUN   TestBatchGet
I0113 08:22:52.213792 998 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.03s)
=== RUN   TestBatchMerge
I0113 08:22:52.244851 998 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.00s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/storage/client_raft_test.go
+++ b/storage/client_raft_test.go
@@ -472,7 +472,6 @@ func TestReplicateAfterTruncation(t *testing.T) {
 	if _, err := client.SendWrapped(rg1(mtc.stores[0]), nil, &incArgs); err != nil {
 		t.Fatal(err)
 	}
-	mvcc := rng.GetMVCCStats()
 
 	// Now add the second replica.
 	if err := rng.ChangeReplicas(roachpb.ADD_REPLICA,
@@ -492,11 +491,11 @@ func TestReplicateAfterTruncation(t *testing.T) {
 		if err != nil {
 			return false
 		}
-		getResp := reply.(*roachpb.GetResponse)
+		val := mustGetInt(reply.(*roachpb.GetResponse).Value)
 		if log.V(1) {
-			log.Infof("read value %d", mustGetInt(getResp.Value))
+			log.Infof("read value %d", val)
 		}
-		return mustGetInt(getResp.Value) == 16
+		return val == 16
 	}, replicaReadTimeout); err != nil {
 		t.Fatal(err)
 	}
@@ -505,8 +504,8 @@ func TestReplicateAfterTruncation(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	if mvcc2 := rng2.GetMVCCStats(); !reflect.DeepEqual(mvcc, mvcc2) {
-		log.Errorf("expected stats on new range to equal old; %+v != %+v", mvcc2, mvcc)
+	if mvcc, mvcc2 := rng.GetMVCCStats(), rng2.GetMVCCStats(); !reflect.DeepEqual(mvcc, mvcc2) {
+		t.Fatalf("expected stats on new range:\n%+v\nto equal old:\n%+v", mvcc2, mvcc)
 	}
 
 	// Send a third command to verify that the log states are synced up so the
@@ -524,9 +523,11 @@ func TestReplicateAfterTruncation(t *testing.T) {
 		if err != nil {
 			return false
 		}
-		getResp := reply.(*roachpb.GetResponse)
-		log.Infof("read value %d", mustGetInt(getResp.Value))
-		return mustGetInt(getResp.Value) == 39
+		val := mustGetInt(reply.(*roachpb.GetResponse).Value)
+		if log.V(1) {
+			log.Infof("read value %d", val)
+		}
+		return val == 39
 	}, replicaReadTimeout); err != nil {
 		t.Fatal(err)
 	}

